packages/authentication/cardstack/middleware.js

```diff
@@ -109,6 +111,40 @@ class Authentication {
     });
   }
 
+  _tokenStatusPreflight(prefix) {
+    return route.options(`/${prefix}/:module/status`,  async (ctxt) => {
+      addCorsHeaders(ctxt.response);
+      ctxt.status = 200;
+    });
+  }
+
+  _tokenStatus(prefix){
+    return route.get(`/${prefix}/:module/status`, compose([
+      async (ctxt) => {
+        addCorsHeaders(ctxt.response);
+        await withJsonErrorHandling(ctxt, async () => {
+          let session = ctxt.state.cardstackSession;
+          if (!session) {
+            ctxt.status = 401;
+            ctxt.body = {
+              errors: [{
+                title: "Not authorized",
+                detail: "The authentication plugin was unable to validate the session"
+              }]
+            };
+            return;
+          }
+          let user = await this.userSearcher.get(session.type, session.id);
+
+          if (!user) { throw new Error(`cant find user type ${session.type} id ${session.id}`); }
+
+          ctxt.status = 200;
+          ctxt.body = user;
```

Comment:
I think this leaks any private fields on the user model (one that the user does not have a grant to see). See `applyReadAuthorization` in `_invokeAuthenticationSource` -- I think we need the same thing here. (And hopefully a test to make sure it remains true).

Reply:
whoops! thanks